rm-datasource/src/main/java/org/apache/seata/rm/datasource/xa/ConnectionProxyXA.java

```diff
@@ -226,7 +226,12 @@ public synchronized void commit() throws SQLException {
             long now = System.currentTimeMillis();
             checkTimeout(now);
             setPrepareTime(now);
-            xaResource.prepare(xaBranchXid);
+            int prepare = xaResource.prepare(xaBranchXid);
+            if (prepare == XAResource.XA_RDONLY
+                    && JdbcConstants.ORACLE.equals(resource.getDbType())) {
```

Comment:
The community meeting has agreed to this approach, and in addition, can you test other databases to see if the corresponding prepare result codes are the same? If so, we hope to remove the special handling of Oracle alone.

Reply:
Sure, purely speaking from MySQL and Oracle, the database type check can be eliminated. I'll try PostgreSQL next, and over the weekend, I'll look for resources to test MSSQL Server.

Reply:
I tested four types of databases: MySQL (8), Oracle (12c), Postgres (16), and MSSQL Server (2022). Only Oracle has read-only optimization; the others do not provide read-only feedback. Based on these four databases, the database type check can be eliminated here.